service/src/test/java/com/kodemon/service/test/TrainerFightServiceTest.java

```diff
@@ -0,0 +1,222 @@
+package com.kodemon.service.test;
+
+import com.kodemon.persistence.dao.TrainerFightDao;
+import com.kodemon.persistence.entity.Gym;
+import com.kodemon.persistence.entity.Pokemon;
+import com.kodemon.persistence.entity.Trainer;
+import com.kodemon.persistence.entity.TrainerFight;
+import com.kodemon.persistence.enums.PokemonName;
+import com.kodemon.persistence.enums.PokemonType;
+import com.kodemon.service.config.ServiceConfig;
+import com.kodemon.service.interfaces.PokemonFightService;
+import com.kodemon.service.interfaces.TrainerFightService;
+import com.kodemon.service.util.Pair;
+import org.hibernate.service.spi.ServiceException;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.dao.DataAccessException;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTransactionalTestNGSpringContextTests;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+import java.util.*;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author Oliver Roch
+ */
+
+@ContextConfiguration(classes=ServiceConfig.class)
+public class TrainerFightServiceTest extends AbstractTransactionalTestNGSpringContextTests {
+    @Mock
+    private PokemonFightService pokemonFightService;
+
+    @Mock
+    private TrainerFightDao trainerFightDao;
+
+    @Inject
+    @InjectMocks
+    private TrainerFightService trainerFightService;
+
+    private Trainer challenger;
+    private Trainer defender;
+
+    private TrainerFight firstFight;
+    private TrainerFight secondFight;
+
+    private Pokemon pikachu;
+    private Pokemon onix;
+
+    private Gym targetGym1;
+    private Gym targetGym2;
+
+    @BeforeClass
+    public void setup() throws ServiceException {
```

Comment:
Same for the `PokemonFightServiceTest`. You said you'd remove it, but it's still there, please fix it.